discretize/BaseMesh.py

```diff
@@ -717,13 +726,20 @@ def nFz(self):
         return int(self.vnFz.prod())
 
     def r(self, x, xType='CC', outType='CC', format='V'):
-        """`r` is a quick reshape command that will do the best it
+        raise DeprecationWarning(
+            'This method will be depreciated in favor if `reshape`. Please use'
+            ' `reshape` instead.'
+        ):
+        return self.reshape(x, xType, outType, format)
+
+    def reshape(self, x, xType='CC', outType='CC', format='V'):
```

Comment:
I think we should also rethink input variable names here (e.g. `format` should not be used). What about 
```python
def reshape(self, x, x_loc='CC', out_loc='CC', out_format='V')
```

I think "loc" is a bit more descriptive (and even "location"). I am still not sure about "format". (It should also take "vec", "vector", "matrix" to let the user be a bit more flexible in the input). 

Thoughts? 

cc @rowanc1